crates/oxc_linter/src/config/config_builder.rs

```diff
@@ -109,23 +109,29 @@ impl ConfigStoreBuilder {
         }
 
         {
-            let all_rules = builder.cache.borrow();
-
             if !extends.is_empty() {
-                let config_file_path = builder.config.path.as_ref().and_then(|p| p.parent());
+                let config_path = builder.config.path.clone();
+                let config_path_parent = config_path.as_ref().and_then(|p| p.parent());
                 for path in &extends {
                     // resolve path relative to config path
-                    let path = match config_file_path {
+                    let path = match config_path_parent {
                         Some(config_file_path) => &config_file_path.join(path),
                         None => path,
                     };
                     // TODO: throw an error if this is a self-referential extend
                     // TODO(perf): use a global config cache to avoid re-parsing the same file multiple times
                     match Oxlintrc::from_file(path) {
                         Ok(extended_config) => {
-                            extended_config
-                                .rules
-                                .override_rules(&mut builder.rules, all_rules.as_slice());
+                            // TODO(refactor): can we merge this together? seems redundant to use `override_rules` and then
+                            // use `ConfigStoreBuilder`, but we don't have a better way of loading rules from config files other than that.
+                            // Use `override_rules` to apply rule configurations and add/remove rules as needed
+                            extended_config.rules.override_rules(&mut builder.rules, &RULES);
```

Comment:
we are not using the RulesCache anymore and instead passing all rules.

Reply:
good point, I was having some borrow checking issues here earlier, but turned out to be easier to resolve than I thought.